pkg/services/ngalert/store/alert_rule.go

```diff
@@ -526,6 +546,18 @@ func (st DBstore) UpdateRuleGroup(cmd UpdateRuleGroupCmd) error {
 				new.Labels = r.ApiRuleNode.Labels
 			}
 
+			if s := new.Annotations["__dashboardUid__"]; s != "" {
```

Comment:
There is constant https://github.com/grafana/grafana/blob/a0dac9c6d9297bf0740a8689f37701f18215f588/public/app/features/alerting/unified/utils/constants.ts#L19 I wonder if it makes sense to use it instead of literal.